extension/src/test/bugs/333.test.ts

```diff
@@ -0,0 +1,271 @@
+import * as assert from "assert";
+import { RefreshXlfHint } from "../../Enums";
+import { RefreshResult } from "../../RefreshResult";
+import { LanguageFunctionsSettings } from "../../Settings/LanguageFunctionsSettings";
+import * as SettingsLoader from "../../Settings/SettingsLoader";
+import { TranslationToken, Xliff } from "../../Xliff/XLIFFDocument";
+import * as XliffFunctions from "../../XliffFunctions";
+
+suite(
+  "#333 After translation statistics show changes although no changes have been made",
+  function () {
+    test("Regression test", function () {
+      const settings = SettingsLoader.getSettings();
+      const lfSettings = new LanguageFunctionsSettings(settings);
+      const gXlf = Xliff.fromString(gXlfXml());
+      const refreshResult = new RefreshResult();
+      const langXlf = XliffFunctions.refreshSelectedXlfFileFromGXlf(
+        Xliff.fromString(langXlfXml()),
+        gXlf,
+        lfSettings,
+        new Map<string, Map<string, string[]>[]>(),
+        refreshResult,
+        false
+      );
+      assert.strictEqual(
+        langXlf.transunit.length,
+        4,
+        "Unexpected number of trans-units."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfUpdatedNotes,
+        0,
+        "Unexpected number of updated notes."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfUpdatedMaxWidths,
+        0,
+        "Unexpected number of updated max-widths"
+      );
+      assert.strictEqual(
+        refreshResult.numberOfUpdatedSources,
+        0,
+        "Unexpected number of updated sources."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfRemovedTransUnits,
+        0,
+        "Unexpected number of removed trans-units"
+      );
+      assert.strictEqual(
+        refreshResult.numberOfRemovedNotes,
+        0,
+        "Unexpected number of removed notes"
+      );
+      assert.strictEqual(
+        refreshResult.numberOfCheckedFiles,
+        0,
+        "Unexpected number of checked files."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfSuggestionsAdded,
+        0,
+        "Did not expect suggestion to be added."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfAddedTransUnitElements,
+        3,
+        "Unexpected number of trans-units added."
+      );
+      assert.strictEqual(
+        refreshResult.numberOfReviewsAdded,
+        5,
+        "Unexpected number of reviews added."
+      );
+
+      const nabFrom = "NAB AL Tool Refresh Xlf";
+      // TransUnit 0
+      assert.strictEqual(
```

Comment:
You are so thorough when doing your asserts below, so you'd might as well compare the complete xlf output! 😅

Reply:
Hopefully I'll thank myself someday in the future :sweat_smile: